[docs] Add health check docs

GavinFrazar opened the pull request:
Part of https://github.com/gravitational/teleport/issues/20544
- https://github.com/gravitational/teleport/issues/20544

Adds docs reference for the `health_check_config` resource and a guide for database health checks.

GavinFrazar committed:
Add health_check_config docs reference

GavinFrazar committed:
Add health check reference details

greedy52 commented on docs/pages/reference/resource-health-checks.mdx:
```diff
@@ -0,0 +1,95 @@
+It includes settings for the health check interval duration, timeout duration, healthy and unhealthy thresholds, and label matchers that determine which resources should be health checked.
+
+`tctl [create | get | edit | rm]` supports `health_check_config` resources (and its alias: `hcc`).
+For example, you can view the default health check configuration using `tctl get`:
```
Could we provide one or more examples that actually uses the settings in the table below and for db_label_expressions.

Should we also mention you can have multiple configs?

greedy52 commented on docs/pages/reference/resource-health-checks.mdx:
```diff
@@ -0,0 +1,95 @@
+
+## View health status
+
+The Teleport web UI will highlight unhealthy resources and provide details about health check failures.
```
should we provide a screenshot?

greedy52 replied:
thanks. nitpick, the screenshot is a little too wide with lots of empty spaces in the middle.

GavinFrazar committed:
add screenshot

GavinFrazar committed:
mention multiple configs, provide example of a more complicated predicate expression

hugoShaka commented on docs/pages/reference/resource-health-checks.mdx:
```diff
@@ -0,0 +1,95 @@
+
+## How it works
+
+Teleport's `health_check_config` API resource determines which resources will have health checks enabled and what settings will be used for those checks.
```
I can rename the command to `kinds`, `resource-kinds`, `kind-list`, or `supported-kinds` if this is confusing. I picked the name to mimick the kube one which is well known by most kube users.

GavinFrazar committed:
Refactor

Refactor the reference into a guide specifically for databases and move
it into the database access guides section.

Adjust the web ui screenshot to be narrower.

GavinFrazar committed:
fix wording around HA routing

GavinFrazar committed:
rename web ui section to troubleshooting and a link

GavinFrazar committed:
add faq

GavinFrazar committed:
fix lint

GavinFrazar committed:
fix wording

GavinFrazar committed:
remove conda env from example

ptgott commented on docs/pages/enroll-resources/database-access/guides/health-checks.mdx:
```diff
@@ -0,0 +1,243 @@
+---
+
+This documentation explains how to manage database health checks.
+Available in Teleport 18, database health checks are used to monitor connectivity from Teleport Database Services to databases enrolled in the Teleport cluster.
```
```suggestion
Available in Teleport 18, database health checks are used to monitor connectivity from Teleport Database Service instances to databases enrolled in a Teleport cluster.
```

ptgott commented on docs/pages/enroll-resources/database-access/guides/health-checks.mdx:
```diff
@@ -0,0 +1,243 @@
+Why monitor database connectivity with health checks?
+
+- **Observability**: Automatically identify networking problems rather than discovering connection issues at user connection time.
+- **High Availability**: Multiple Teleport Database Services can be deployed to proxy connections to the same enrolled databases. Teleport prioritizes routing user connections through Database Services that can reach the database endpoint over those that cannot.
```
```suggestion
- **High Availability**: Multiple Teleport Database Service instances can be deployed to proxy connections to the same enrolled databases. Teleport prioritizes routing user connections through Database Service instances that can reach the database endpoint over those that cannot.
```

ptgott reviewed (commented):
Mostly minor things, with one structure question

GavinFrazar committed:
Update docs/pages/enroll-resources/database-access/guides/health-checks.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

GavinFrazar committed:
Update docs/pages/enroll-resources/database-access/guides/health-checks.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

GavinFrazar committed:
Update docs/pages/enroll-resources/database-access/guides/health-checks.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

GavinFrazar committed:
Elaborate a bit on observability

GavinFrazar committed:
Database Services -> Database Service instances
